openhcl/virt_mshv_vtl/src/processor/mod.rs

```diff
@@ -897,20 +896,8 @@ impl<'a, T: Backing> UhProcessor<'a, T> {
 
                     if vtl == GuestVtl::Vtl1 {
                         assert!(self.partition.isolation.is_hardware_isolated());
-                        // Should not have already initialized the hv emulator for this vtl
-                        assert!(self.hv[vtl].is_none());
-
-                        self.hv[vtl] = Some(
-                            self.partition
-                                .hv
-                                .as_ref()
-                                .expect("has an hv emulator")
-                                .add_vp(
-                                    self.partition.gm[GuestVtl::Vtl1].clone(),
-                                    self.vp_index(),
-                                    Vtl::Vtl1,
-                                ),
-                        );
+                        // Should have already initialized the hv emulator for this vtl
+                        assert!(self.backing.hv(vtl).is_some());
```

Comment:
Losing this bit of validation is unfortunate, but how necessary is it? Can probably do a bit more wiring to keep it around if need be.

Reply:
What validation did we lose?

Reply:
Anything that attempts to access the VTL 1 hv before it is enabled would previously fail on an unwrap or similar. Now they'd succeed.

Reply:
Oh, the whole thing where we lazily create the emulator? That seems OK to me.